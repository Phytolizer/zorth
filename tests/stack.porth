include "std.porth"

// push int
10 20 30 print print print
10 print 20 print 30 print

"------------------------------\n" stdout write

// push str
"Hello, World\n" stdout write // ASCII
"Привет, Мир!\n" stdout write // UTF-8

"------------------------------\n" stdout write

// characters
'E' print

"------------------------------\n" stdout write

// dup
420 dup dup print print print
69 dup print dup print print

"------------------------------\n" stdout write

// 2dup
69 420 2dup print print print print

"------------------------------\n" stdout write

// swap
69 420 swap print print

"------------------------------\n" stdout write

// drop
69 420 drop print

"------------------------------\n" stdout write

// over
69 420 over print

"------------------------------\n" stdout write

// big number
1234567890987654321 print
