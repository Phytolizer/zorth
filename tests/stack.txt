30
20
10
10
20
30
------------------------------
Hello, World
你好世界！
你好世界！
------------------------------
420
420
420
69
69
69
------------------------------
420
69
420
69
------------------------------
69
420
------------------------------
69
------------------------------
69
420
69
------------------------------
1234567890987654321
