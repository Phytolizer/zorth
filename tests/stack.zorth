// push
10 20 30 dump dump dump
10 dump 20 dump 30 dump

// dup
420 dup dup dump dump dump
69 dup dump dup dump dump
