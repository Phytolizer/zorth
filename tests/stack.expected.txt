30
20
10
10
20
30
111111111
420
420
420
69
69
69
111111111
420
69
420
69
111111111
69
420
111111111
69
111111111
69
